### DISCUSSION TITLE
Curso Fullstack > Documentando a API > Testando a API

### DISCUSSION THREAD

**Discussion by VictorTine** [balta-io/forum#631] [Tira Dúvidas] [answered]
Segue erro no print ao tentar rodar o projeto
![Captura de tela 2024-07-20 182317](https://github.com/user-attachments/assets/33b93f25-8ce6-48e8-8c37-65a18d9d2b80)

**Comment by VictorTine** [comment]
Fiz algumas correções no código. Tirei os parâmetros do AddSwaggerGen(); e corrigi a sintaxe de WithName.

Porém continuo com erro para rodar

![image](https://github.com/user-attachments/assets/46b5f402-3194-4a8b-9212-749ff9638aea)

**Comment by VictorTine** [comment]
Atualmente estou com essa versão do código, mas sigo com erro.

![image](https://github.com/user-attachments/assets/287f8720-e37e-47d2-8148-7a053ffa2f53)

**Comment by VictorTine** [comment]
![image](https://github.com/user-attachments/assets/5676e2a4-b4a1-46c8-a114-975d05bee9a5)

Após algumas tentativas, esse é o código mais atualizado, e o erro que o VS informa

**Reply by andrebaltieri** [reply]
Boa noite, @VictorTine, como vai?

Pelo que vi nos seus códigos, você está tentando usar em diversos locais o `:`, porém o código correto não tem este trecho.
![image](https://github.com/user-attachments/assets/e1e75575-eba8-4801-8cbe-eea9d938757a)

O código acima aparece `:SwaggerGenOptions` e `:Type` apenas como uma amostra dos tipos, isto é colocado pela IDE.
O código original fica como mostro nos vídeos:

```csharp
builder.Services.AddEndpointsApiExplorer();
builder.Services.AddSwaggerGen(x => { x.CustomSchemaIds(n => n.FullName); });
```

**Reply by VictorTine** [reply]
Olá Balta, agradeço o apoio.

Tirei os ":" que aparecem em tela na aula por conta da IDE, mas mesmo assim continuo sem conseguir rodar.

Segue o Erro:
![image](https://github.com/user-attachments/assets/cce81bc7-397f-4bed-81b4-07e8fc304fb3)
E mais um pouco do código para contexto
![image](https://github.com/user-attachments/assets/ea2f8d0f-95bf-47bf-862d-614e9b743fe8)
![image](https://github.com/user-attachments/assets/77d26c9b-24df-4471-9b49-099d7113f216)

Não sei se configurei algo errado, estou pensando em começar tudo novamente do zero.

**Comment by FBODeveloper** [comment, marked as answer]
"builder.Services.AddTransient<Handler>();"

Faltou isso....
não sei pq o github esconde o Handler no texto acima...... AddTransientMENORHandlerMAIOR()

tá nessa aula
https://balta.io/player/assistir/c4bfdf26-8709-46f4-b918-d59e78953e4a/cdbef730-a7dd-4e2d-ac69-12d0d9e14e47

**Reply by VictorTine** [reply]
Agradeço pela ajuda cara. Vi a aula toda e adicionei o comando faltante. Antes também estava com erro pq coloquei n => n.Fullname e não n => n.FullName. Tava com esse erro de sintaxe tbm.
